src/nap.js

```diff
@@ -157,7 +157,9 @@ function bySelector(){
 }
 
 function wrap(fn, stack) {
-  return stack.reduce(middleware, fn)
+  return function wrapped() {
+    stack.reduce(middleware, fn).apply(null, arguments)
+  }
```

Comment:
The previous implementation returned a function, that built a composition from the middlewares and calling the handler at the end with the result from the composition. In the new implementation a function is returned, which composes the same list when it is called, which means that all the *current* middlewares will be called, not just the ones that were added with `use` before the composition is created.